main.go

```diff
@@ -97,14 +108,14 @@ func main() {
 	usernames = make(map[string]string)
 
 	for userID, _ := range c.Users {
-		if _, ok := usernames[userID]; !ok {
+		usernames[strings.ToLower(c.Users[userID].Nick)] = userID
+		if _, ok := notifications[userID]; !ok {
 			notifications[userID] = make(map[string]Notification)
```

Comment:
I don't understand why you need to initialize notifications with an empty map here, but I guess it would be uglier to do it when needed further down the code path. Again with a new `Notifications` type it could be done automatically when something is added to the map.